### ISSUE TITLE
Fast scroll bug

### ISSUE THREAD

**Issue by velthune** [uacaps/PageMenu#46]
Hi, I'm testing your project and I think is nice, really.

But there is a little bug: when I scroll faster (left/right is the same) background color is presented.

Please try this test to reproduce the problem:
- tap down/up scrolling left
- as soon you have pull up your finger, tap again and keep your finger on device
- now try to scroll left, always keeping finger on device

For the moment I fix with this workaround:

    func willMoveToPage(controller: UIViewController, index: Int) {
        pageMenu?.view.userInteractionEnabled = false;
    }
    
    func didMoveToPage(controller: UIViewController, index: Int) {
        pageMenu?.view.userInteractionEnabled = true;
    }

**Comment by nagaho** [comment]
+1

**Comment by fahlout** [comment]
It's a known issue, but I unfortunately haven't figured out a good fix for it yet

**Comment by nagaho** [comment]
> oldVC.view.removeFromSuperview()
> oldVC.removeFromParentViewController()

Any problems with that remove there method in func removePageAtIndex?

**Comment by fahlout** [comment]
Yes, I know why the problem occurs. I am trying to only have one vc in memory at a time which causes this behavior.

**Comment by jyounus** [comment]
How about keeping up to 3 VCs in memory? The currently active one, the tab on the left and the one on the right. Similar to how tableview cells work, having the ones visible on the screen and one extra at the top and bottom. Obviously if you only have 3 tabs, this would be pointless. But if you had anything above 3, it would still be good. 

Maybe have a properly to allow devs to easily modify this behaviour and change this if they want to only load one VC at a time? This would allow flexibility on older devices where testing shows having more than 1 VC in memory is causing performance issues.

Idk, just a suggestion.

**Comment by fahlout** [comment]
This was the plan at the beginning of development on this, but I got into some road blocks trying to get 3 VCs to be loaded at once, which would most likely prevent this problem and still run fine on newer devices. having both options would be nice. Unfortunately, though, I don't have time to work on this at the moment, but if you are interested in looking into this, I would definitely appreciate your work. Otherwise, it'll have to wait for a while..

**Comment by jyounus** [comment]
I'm quite busy myself at the moment with other stuff but if I do get some time, I will attempt it.

**Comment by btxkenshin** [comment]
hi @fahlout , i have commit a pull request to fix the issues.
just as @jyounus suggests:
How about keeping up to 3 VCs in memory? The currently active one, the tab on the left and the one on the right

[link](https://github.com/uacaps/PageMenu/pull/74)

hope to check it~
